wikimedia/wikipedia-ios
# Attempt to fix WMFAlertManager dismiss race condition

**Phabricator:** 
https://phabricator.wikimedia.org/T323517

### Notes
While investigating this to see if we can push it out until our new toasts design component, I tried a minor fix that seems to work in my testing.

Even though the alerts displayed on PageHistoryViewController.swift specifically set `dismissPreviousAlerts` = true, we still had stacked toasts in the case of no internet connection. This is due to multiple error alerts firing at once, because we are hitting many endpoints on that page.

I suspect it had something to do with a race condition in this area. Adjusting it so that alert blocks are queued seems to fix it. I'm ensuring the array it's read from and written to on the main thread, so it should be thread-safe.

### Test Steps
1. Visit an article. Turn on airplane mode.
2. Go to Edit History
3. Confirm you only see one "No internet connection" banner instead of 2 or 3.
